drwmutex.go

```diff
@@ -122,11 +136,15 @@ func (dm *DRWMutex) lockBlocking(isReadLock bool) {
 				copy(dm.writeLocks, locks[:])
 			}
 
-			return
+			return true
+		}
+		if time.Since(start) >= timeout { // Are we past the timeout?
```

Comment:
Above `success := lock(clnts, &locks, dm.Name, isReadLock)` may run beyond given `timeout`.   This means this method is not guarantied to run with given `timeout` ie if my timeout is one second, i might get back control after 10 seconds as well.

Is it what we wanted?